### PR TITLE
fix(eslint): manual fix of eslint errors on packages @o3r/{l,m}*

### PR DESCRIPTION
**Pull request by matthieu-crouzet**

### CODE REVIEW

File: eslint.shared.config.mjs [hunk 1]
```diff
@@ -37,6 +37,7 @@ export default [
         ...globals.browser,
         ...globals.node,
         ...globals.webextensions,
+        NodeJS: true,
```
Comment (kpanot) [inline eslint.shared.config.mjs:40]: This is the case for `builders`, `cli`, `schematics`, `test` and `scripts`.
It may be very valuable to forbid `Node` in the other cases.
(same comment for `globals.node` and reverse comment for `globals.browser`)

File: packages/@o3r/logger/src/services/logger/logger.spec.ts [hunk 2]
```diff
@@ -102,60 +121,81 @@ describe('Logger service', () => {
...
         client1 = {
           identify: jest.fn(),
           event: jest.fn(),
           getSessionURL: jest.fn().mockReturnValue('client.session.url1'),
-          stopRecording: jest.fn(),
-          resumeRecording: jest.fn(),
-          error: jest.fn(),
-          warn: jest.fn(),
-          log: jest.fn(),
+          stopRecording: stopRecordingMock1,
```
Comment (kpanot) [inline packages/@o3r/logger/src/services/logger/logger.spec.ts:183]: Is it coming from a Rule?
This change is fine but a mockReset instead of a new instance can lead to issue if not correctly applied

File: eslint.shared.config.mjs (hunk 1 above)
Comment (matthieu-crouzet) [reply eslint.shared.config.mjs:40]: It's only for us, it's not the shared config

File: packages/@o3r/logger/src/services/logger/logger.spec.ts (hunk 2 above)
Comment (matthieu-crouzet) [reply packages/@o3r/logger/src/services/logger/logger.spec.ts:183]: Most of the time it's to fix [@typescript-eslint/unbound-method](https://typescript-eslint.io/rules/unbound-method/) 
I run the test in local for this package and they are still green :)

File: eslint.shared.config.mjs (hunk 1 above)
Comment (kpanot) [reply eslint.shared.config.mjs:40]: yes, my comment was for us :)

File: packages/@o3r/logger/src/services/logger/logger.spec.ts (hunk 2 above)
Comment (kpanot) [reply packages/@o3r/logger/src/services/logger/logger.spec.ts:183]: I knew the tests were fine.
The `jest.fn()` is not part of the exceptions handled by `eslint-plugin-jest` (as described in the linked doc)?

Comment (matthieu-crouzet) [reply packages/@o3r/logger/src/services/logger/logger.spec.ts:183]: issue is not on `jest.fn()` but on `expect(client.log)` because we gave the reference to an unbound method

Comment (matthieu-crouzet) [reply packages/@o3r/logger/src/services/logger/logger.spec.ts:183]: I got your point 
I was not aware this rule [jest/unbound-method](https://github.com/jest-community/eslint-plugin-jest/blob/HEAD/docs/rules/unbound-method.md)
This rule is not activated on our config 
we may check if we want to activate it or not

File: eslint.shared.config.mjs (hunk 1 above)
Comment (matthieu-crouzet) [reply eslint.shared.config.mjs:40]: I take the point 
I will do it in another PR

File: packages/@o3r/localization/src/tools/localization.service.ts [hunk 3]
```diff
@@ -49,7 +48,7 @@ export class LocalizationService {
   /**
    * Internal subject that we use to track changes between keys only and translation mode
    */
-  // eslint-disable-next-line @typescript-eslint/naming-convention
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- private field
   private readonly _showKeys$ = new BehaviorSubject(false);
```
Comment (vscaiceanu-1a) [inline packages/@o3r/localization/src/tools/localization.service.ts:52]:
```suggestion
  private readonly showKeys = new BehaviorSubject(false);
```

Comment (matthieu-crouzet) [reply packages/@o3r/localization/src/tools/localization.service.ts:52]: One of our recommendation is to have a suffix `$` when using observable or subject no ?

File: eslint.shared.config.mjs (hunk 1 above)
Comment (matthieu-crouzet) [reply eslint.shared.config.mjs:40]: https://github.com/AmadeusITGroup/otter/pull/2361

File: packages/@o3r/localization/schematics/ng-add/index.ts [hunk 4]
```diff
@@ -72,6 +72,9 @@ function ngAddFn(options: NgAddSchematicsSchema): Rule {
       };
     });
     const registerDevtoolRule = await registerDevtools(options);
+    const schematicDefaultOptions = Object.fromEntries(['@o3r/core:component*'].map((name) => [name, {
```
Comment (fpaul-1A) [inline packages/@o3r/localization/schematics/ng-add/index.ts:75]: are we really expecting more entries? or is it just a trick to make the linter happy?

Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/ng-add/index.ts:75]: For some schematics we have more entries 
So I did it to have the same format everywhere I saw it

File: packages/@o3r/localization/src/tools/localization.service.ts (hunk 3 above)
Comment (vscaiceanu-1a) [reply packages/@o3r/localization/src/tools/localization.service.ts:52]: I don't think we recommend this for Subject, only for Observables.

File: packages/@o3r/localization/builders/helpers/localization.generator.ts [hunk 5]
```diff
@@ -226,11 +226,12 @@ export class LocalizationExtractor {
...
   public getPatternsFromTsConfig() {
-    const tsconfigResult = ts.readConfigFile(this.tsconfigPath, ts.sys.readFile);
+    const tsconfigResult = ts.readConfigFile(this.tsconfigPath, (p) => ts.sys.readFile(p));
 
     if (tsconfigResult.error) {
-      // eslint-disable-next-line @typescript-eslint/no-base-to-string
-      const stringError = tsconfigResult.error.messageText.toString();
+      const stringError = typeof tsconfigResult.error.messageText === 'string'
+        ? tsconfigResult.error.messageText
+        : tsconfigResult.error.messageText.messageText;
```
Comment (fpaul-1A) [inline packages/@o3r/localization/builders/helpers/localization.generator.ts:234]: is this really equivalent to `tsconfigResult.error.messageText.toString()`?

File: packages/@o3r/localization/schematics/localization-base/index.ts [hunk 6]
```diff
@@ -463,11 +469,6 @@ export function updateLocalization(options: { projectName?: string | null | unde
   ]);
 }
 
-/**
- *
- * @param options
- * @param options.projectName
- */
 function updateI18nFn(options: { projectName?: string | undefined }): Rule {
```
Comment (sdo-1A) [inline packages/@o3r/localization/schematics/localization-base/index.ts:472]: why did you remove the jsdoc ?

File: packages/@o3r/mobile/src/capacitor/helpers.spec.ts [hunk 7]
```diff
@@ -9,9 +9,6 @@ import {
   isCapacitorContext
 } from './helpers';
 
-/**
- * @param platform the device platform: ios, android or web
- */
 function setCapacitorPlatform(platform: CapacitorPlatforms) {
```
Comment (sdo-1A) [inline packages/@o3r/mobile/src/capacitor/helpers.spec.ts:12]: why did you remove the jsdoc ?

File: packages/@o3r/localization/builders/helpers/localization.generator.ts (hunk 5 above)
Comment (matthieu-crouzet) [reply packages/@o3r/localization/builders/helpers/localization.generator.ts:234]: if `tsconfigResult.error.messageText` was not of type string it will return `[object object]`, no ?
that's what the rule said

File: packages/@o3r/localization/schematics/localization-base/index.ts (hunk 6 above)
Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/localization-base/index.ts:472]: It's a private function so jsdoc is not mandatory and this doc was not adding any values

File: packages/@o3r/localization/src/tools/localization.service.ts (hunk 3 above)
Comment (kpanot) [reply packages/@o3r/localization/src/tools/localization.service.ts:52]: Indeed the private field name can be changed (or the rule adapted) to not disable `no-underscore-dangle` in several places

Comment (matthieu-crouzet) [reply packages/@o3r/localization/src/tools/localization.service.ts:51]: Fixed with the new config

File: packages/@o3r/logger/src/services/logger/logger.spec.ts (hunk 2 above)
Comment (matthieu-crouzet) [reply packages/@o3r/logger/src/services/logger/logger.spec.ts:183]: We will see later if we can/should use [jest/unbound-method](https://github.com/jest-community/eslint-plugin-jest/blob/HEAD/docs/rules/unbound-method.md)

File: packages/@o3r/logger/src/services/logger/logger.spec.ts [hunk 8]
```diff
@@ -60,18 +60,37 @@ describe('Logger service', () => {
...
+      identifyMock.mockReset();
+      eventMock.mockReset();
+      getSessionURLMock.mockReset();
+      stopRecordingMock.mockReset();
+      resumeRecordingMock.mockReset();
+      errorMock.mockReset();
+      warnMock.mockReset();
+      logMock.mockReset();
+      createMetaReducerMock.mockReset();
+      getSessionURLMock.mockReturnValue('client.session.url');
```
Comment (fpaul-1A) [inline packages/@o3r/logger/src/services/logger/logger.spec.ts:83]: that seems like a lot of boilerplate just to please the linter

File: packages/@o3r/localization/schematics/ng-update/v10-0/index.ts [hunk 9]
```diff
@@ -36,4 +36,5 @@ function updateV10_0Fn(): Rule {
 /**
  * Update of Otter library V10.0
  */
+// eslint-disable-next-line @typescript-eslint/naming-convention -- contains the version number
```
Comment (mrednic-1A) [inline packages/@o3r/localization/schematics/ng-update/v10-0/index.ts:39]: is it not better to have it at the top of the file ?
We might still add migration functions.

File: packages/@o3r/logger/src/services/logger/logger.spec.ts (hunk 8 above)
Comment (matthieu-crouzet) [reply packages/@o3r/logger/src/services/logger/logger.spec.ts]: no need of changes with `jest/unbound-method`

File: packages/@o3r/localization/schematics/ng-update/v10-0/index.ts (hunk 9 above)
Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/ng-update/v10-0/index.ts:39]: That's something we discussed with the team and we said that it's better to disable only where it's needed
If you want we can re-discussed it to set a global config on the repo to disable this rule for all files inside an `schematics/ng-update` folder

**Review by fpaul-1A — APPROVED** [review]

**Review by mrednic-1A — APPROVED** [review]
